docs/tutorial_add_executor.rst

```diff
@@ -0,0 +1,417 @@
+            "job_state":"F",
+            "comment":"Job run at Mon Jan 24 at 08:39 on (edtb-01[0]:ncpus=1) and finished",
+            "Exit_status":0,
+        }
+    }
+ }
+
+Here is an implementation for ``parse_status_output``, as well as a helper dictionary ``_STATE_MAP``::
+
+    import json
+    from psij import JobState, JobStatus
+    from psij.executors.batch.batch_scheduler_executor import check_status_exit_code
+
+    _STATE_MAP = {
+        'Q': JobState.QUEUED,
+        'R': JobState.ACTIVE,
+        'F': JobState.COMPLETED
+    }
+
+    class PBSProJobExecutor: ...
+
+        def parse_status_output(self, exit_code: int, out: str) -> Dict[str, JobStatus]:
+            check_status_exit_code(_QSTAT_COMMAND, exit_code, out)
+            r = {}
+
+            report = json.loads(out)
+            jobs = report['Jobs']
+            for native_id in jobs:
+                native_state = jobs[native_id]["job_state"]
+                state = _STATE_MAP(native_state)
+
+                msg = jobs[native_id]["comment"]
+                r[native_id] = JobStatus(state, message=msg)
+
+            return r
+
+``parse_status_output`` is given both the stdout and the exit code of ``qstat`` and must either transcribe that into a dictionary of :py:class:`JobStatus` objects describing the state of each job, or raise an exception.
+
+This implementation uses a helper, :py:meth:`check_status_exit_code`, which will raise an exception if ``qstat`` exited with a non-zero exit code. Then, it assumes that the ``qstat`` output is JSON and deserialises, and for each job in the JSON, it uses two fields to create a ``JobStatus`` object: a human readable message is taken from the PBS ``comment`` field, and a machine readable status is converted from a single letter PBS status (such as F for finished, or Q for queued) into a PSI/J :py:class:`JobState` via the ``_STATE_MAP`` dictionary.
+
+With these status methods in place, the ``pytest`` command from before should execute to completion.
+
+We still haven't implemented the cancel methods, though. That will be revealed by running a broader range of tests::
+
+    PYTHONPATH=$PWD/src:$PYTHONPATH pytest 'tests' --executors=pbspro
+
+which should give this error (amongst others -- this commandline formation is ugly and I'd like it to work more along the lines of `make test`)::
+
+    FAILED tests/test_executor.py::test_cancel[pbspro] - NotImplementedError
+
+Implementing cancel
+===================
+
+The two methods to implement for cancellation follow the same pattern as for submission and status:
+
+* :py:meth:`get_cancel_command` - this should form a command for cancelling a job.
+* :py:meth:`process_cancel_command_output` - this should interpret the output from the cancel command.
+
+It looks like you don't actually need to implement process_cancel_command_output beyond the stub we already have, to make the abstract class mechanism happy. Maybe that's something that should change in psi/j?
```

Comment:
You also don't need to implement other error handling routines to make some things work, but I don't think that's a reason not to implement proper error handing.

The specific purpose of this method is explained in the docs. Calling `cancel()` on a completed job is not an error in PSI/J, since the spirit of `cancel()`, to ensure that the job is not running, is fulfilled. But that's not what all queuing systems do. Some will fail when given the job id of a finished job. So this method needs to look at why `qdel/scancel/etc.` has failed, if it has, and react accordingly.

Reply:
Point being, you can update the tutorial to reflect that, or you can remove this sentence, but I don't think this can or should change in PSI/J.